core/src/components/modal/modal.tsx

```diff
@@ -422,24 +423,14 @@ export class Modal implements ComponentInterface, OverlayInterface {
    */
   @Method()
   async present(): Promise<void> {
+    const unlock = await this.lockController.lock();
+
     if (this.presented) {
       return;
```

Comment:
```suggestion
      unlock();
      return;
```

We need to release the lock otherwise the overlay will get deadlocked